### PR TITLE
GitHub Actions: Enable mysql + pgsql services

### PR DESCRIPTION
**Pull request by iredmail**
- Enable mysql + postgresql services for testing.
- Install required SQL drivers to trigger related unittests.
- Specify sql driver versions.

I think we can completely replace Travis CI by GitHub Action now.
Maybe it's time to tag a new release, then entirely drop py2 support.

### CODE REVIEW

File: .github/workflows/lint_python.yml [hunk 1]
```diff
@@ -47,8 +44,17 @@ jobs:
...
+        env:
+          PGPASSWORD: postgres
+
+      - run: createuser -d -P 'tiger' webpy
+        env:
+          PGPASSWORD: postgres
+
+      - run: psql -h localhost -U postgres -d webpy -c "ALTER DATABASE webpy OWNER TO scott;"
+        env:
+          PGPASSWORD: postgres
```
Comment (cclauss) [inline .github/workflows/lint_python.yml]:
```
- run: |
    createdb -h localhost -U postgres webpy
    createuser -d -P 'tiger' webpy
    psql -h localhost -U postgres -d webpy -c "ALTER DATABASE webpy OWNER TO scott;"
  env:
    PGPASSWORD: postgres
```

File: .github/workflows/lint_python.yml [hunk 2]
```diff
@@ -51,5 +37,25 @@ jobs:
...
+          createdb -h localhost -U postgres webpy
+          createuser -h localhost -U postgres -d scott
+          psql -h localhost -U postgres -d postgres -c "ALTER USER scott WITH ENCRYPTED PASSWORD 'tiger'"
+          psql -h localhost -U postgres -d postgres -c "ALTER DATABASE webpy OWNER TO scott"
+        env:
+          PGPASSWORD: postgres
+
+      # Run pytest and get detailed output for easy debugging if test failed.
+      # Env variables `WEBPY_DB_` are required for sql db connections.
+      - run: pytest -s -x -v .
```
Comment (cclauss) [inline .github/workflows/lint_python.yml]: What are the cryptic options s, x, and v...  Do they have alternatives that are more self-explanatory?

File: test_requirements.txt [hunk 3]
```diff
@@ -1,5 +1,14 @@
...
-mysql-connector>=2.1.4
-psycopg2
-PyMySQL
 pytest>=4.6.2
+
+# DB drivers.
+# mysql
+MySQL-python>=1.2.5; python_version == '2.7'
+PyMySQL>=0.9.3
+mysql-connector-python>=8.0.19
```
Comment (cclauss) [inline test_requirements.txt]: Do we really need all three of these?

File: test_requirements.txt [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+
+# pgsql
+#psycopg; python_version == '2.7'
+psycopg2>=2.8.4
+#pgdb>=0.0.8
```
Comment (cclauss) [inline test_requirements.txt]: Can we lose commented lines 12 ans 14?

File: tests/test_db.py [hunk 5]
```diff
@@ -48,9 +48,24 @@ def setup_database(dbname, driver=None, pooling=False):
...
+        user = os.getenv("WEBPY_DB_USER")
+        # Travis CI hack:
+        if not user:
+            user = os.getenv("USER")
+        pw = os.getenv("WEBPY_DB_PASSWORD", "")
+
         db = web.database(
-            dbn=dbname, db="webpy", user=user, pw="", pooling=pooling, driver=driver
+            dbn=dbname,
+            host=host,
```
Comment (cclauss) [inline tests/test_db.py]:
```suggestion
            host=os.getenv("WEBPY_DB_HOST", "localhost"),
```

File: tests/test_db.py [hunk 6]
```diff
@@ -48,9 +48,24 @@ def setup_database(dbname, driver=None, pooling=False):
...
+            port=port,
```
Comment (cclauss) [inline tests/test_db.py]:
```suggestion
            port=os.getenv("WEBPY_DB_PG_PORT", 5432),
```

File: tests/test_db.py [hunk 7]
```diff
@@ -48,9 +48,24 @@ def setup_database(dbname, driver=None, pooling=False):
...
+            db=_dbname,
```
Comment (cclauss) [inline tests/test_db.py]:
```suggestion
            db=os.getenv("WEBPY_DB_NAME", "webpy"),
```

File: tests/test_db.py [hunk 8]
```diff
@@ -48,9 +48,24 @@ def setup_database(dbname, driver=None, pooling=False):
...
+            user=user,
```
Comment (cclauss) [inline tests/test_db.py]:
```suggestion
            user=os.getenv("WEBPY_DB_USER") or os.getenv("USER"),
```

File: tests/test_db.py [hunk 9]
```diff
@@ -48,9 +48,24 @@ def setup_database(dbname, driver=None, pooling=False):
...
+            pw=pw,
```
Comment (cclauss) [inline tests/test_db.py]:
```suggestion
            pw=os.getenv("WEBPY_DB_PASSWORD", ""),
```

File: test_requirements.txt (hunk 3 above)
Comment (iredmail) [reply test_requirements.txt]: Yes.

Check file `tests/test_db.py`, unittests are triggered if the db module is available / importable.

File: test_requirements.txt (hunk 4 above)
Comment (iredmail) [reply test_requirements.txt]: Not yet.

web.py supports them, but seems unitests for`psycopg` and `pgdb` failed. I had to disable them temporarily. After merged this PR, we should enable them for testing again and fix the failed tests.

File: tests/test_db.py (hunk 5 above)
Comment (iredmail) [reply tests/test_db.py]: Fixed.

File: .github/workflows/lint_python.yml (hunk 2 above)
Comment (iredmail) [reply .github/workflows/lint_python.yml]: Fixed.

**Comment by iredmail** [comment]
Nice because based on your nice work. :)

**Comment by anandology** [comment]
Nice!

@iredmail could you please cleanup the commit history before merging?

**Comment by cclauss** [comment]
Before we release again, can we find a solution for #598, #604 and land #603?

File: test_requirements.txt (hunk 4 above)
Comment (anandology) [reply test_requirements.txt]: I think we can mark them as deprecated and drop the support for them in the subsequent release. When web.py started there all these libraries were in use. Not I guess psycogp2 is the only library that everyone uses.

Comment (iredmail) [reply test_requirements.txt]: Both psycopg and pgdb can not even pass tests. There’s no need to wait for one more release to drop them.

**Pull request merged**